### PR TITLE
As/tls protocol

### PR DESCRIPTION
**Pull request by soncuteanca**
# Description

Verify that TLS v1.2 protocol is handled correctly and returns the proper information.

## Bugzilla bug ID

**Testrail: https://testrail.stage.mozaws.net/index.php?/cases/view/192739**
**Link: https://bugzilla.mozilla.org/show_bug.cgi?id=1905188**

## Type of change

Please delete options that are not relevant.

- [x ] New Test

# How does this resolve / make progress on that bug?
Completed

Please describe the progress or significance with respect to the bug listed above.

# Screenshots / Explanations

Please upload any relevant media or add a relevant description with respect to the bug listed above.

# Comments / Concerns

Please add a short blurb about any comments or concerns that this change might cause.

### CODE REVIEW

File: tests/security_and_privacy/test_tls_v1_2_protocol.py
Comment (ben-c-at-moz) [inline tests/security_and_privacy/test_tls_v1_2_protocol.py:44]: So for some reason, the info I get on my machine when I go to the site is:

`Connection Encrypted (TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384, 256 bit keys, TLS 1.2)`, the difference being `RSA` instead of `ECDSA`. We should figure out what's going on here before we merge.

Comment (soncuteanca) [reply tests/security_and_privacy/test_tls_v1_2_protocol.py]: It looks like I've made an oversight. The string mentioned in the C192739 test case is indeed the one with RSA. I have verified this on two different test machines (nightly and beta), and I receive the correct return. However, something at the level of my work laptop gives a different string when run either from the IDE or manually.

I am waiting for a response from a developer, that covers this area and will provide an update once I have more information. For now, I did a commit with the expected string from my personal laptop.

Comment (soncuteanca) [reply tests/security_and_privacy/test_tls_v1_2_protocol.py]: I just had a VPN left in background from my manual testing credential management work. All good after I've disabled it. The test passes locally on my end too with the RSA  string.

**Review by ben-c-at-moz — APPROVED** [review]
LGTM

**Pull request merged**
